### ISSUE TITLE
Tracking issue for Encoder support

### ISSUE THREAD

**Issue by tanekere** [HaoboGu/rmk#69]
Hi 👋,
I wanted to build a project like this one when i found yours. I am not new to rust or embedded programming but am new to embedded programming in Rust. Would this be a good first issue for me?

**Comment by HaoboGu** [comment]
Yes, thanks for raising the issue! You can post anything about encoder support here:D

**Comment by tanekere** [comment]
Took a look into this today and looks like most rotary encoders rely on pull-up inputs. But as far as i can tell the rmk-macro crate can only support pull down inputs as of right now. So i guess that will be the first task. 
Also, the macros just refer to them as 'input_pin' (eg.`convert_gpio_str_to_input_pin`) so should i change names to pd_input or keep it like this and just add functions for pull up?

**Comment by azzamsa** [comment]
Wow, `sm_td` was mentioned. I'm always afraid of leaving `sm_td` behind, as I relied on this after I migrated from Miryoku (but I keep the exact layout). As an avid Rust programmer myself, I can't wait to try `rmk` HRM for my next board. Hopefully, this week!

**Comment by HaoboGu** [comment]
I've already added rotary encoder support, it's available by using Rust code only. I'll add the config soon. Please checkout the example [here](https://github.com/HaoboGu/rmk/blob/main/examples/use_rust/nrf52840_ble/src/main.rs).

**Comment by HaoboGu** [comment]
Close as the rotary encoder support is considered finished
